datafusion/expr/src/expr.rs

```diff
@@ -279,7 +279,7 @@ use sqlparser::ast::{
 #[derive(Clone, PartialEq, PartialOrd, Eq, Debug, Hash)]
 pub enum Expr {
     /// An expression with a specific name.
-    Alias(Alias),
+    Alias(Box<Alias>),
```

Comment:
Another change that might be a bit less impactful would be to box the fields of Alias instead

So Alias {
  expr: Box<Expr>
..
}
```

It may be just as bad / worse though